deploy-board/deploy_board/static/js/components/capacitycomponents.js

```diff
@@ -188,23 +188,66 @@ Vue.component('in-rolling-alert', {
     props:['actionlink']
 });
 
+function getCapacityAlertMessage(isWarning, remainingCapacity, placements, increase) {
+    const errorMessage = `Insufficient combined remaining capacity in this cluster/auto scaling group. `;
+    const instruction = `You can attach additional placements to the corresponding clutter to increase` +
+                        ` total potential capacity at Cluster Configuration -> Advanced Settings.\n`;
+    const status = `Combined remaining capacity: ${remainingCapacity}\n` +
+                   `Current placement(s): ${JSON.stringify(placements, null, 2)}`;
+
+    if (isWarning) {
+        return errorMessage + `You can save this configuration but the cluster/ASG might run into capacity issues in the future. ` +
+                instruction + `Requested size increase: ${increase}\n` + status
+    } else {
+        // error
+        return errorMessage + `You shouldn't save this configuration because the cluster/ASG will run into capacity issues. ` +
+                instruction + `Requested size increase: ${increase}\n` + status;
+    }
+}
 
 /**
  * The capacity button. This is shown when autoscaling is disabled or min\max size are the same.
  * In this case, only one capacity box is shown
  */
 Vue.component("static-capacity-config", {
-    template: '<div class="form-group">\
-    <label for="capacity" class="deployToolTip control-label col-xs-4" title="Number of hosts for this service">\
-        Capacity\
-    </label>\
-    <div class="col-xs-2" >\
-    <input class="form-control" v-bind:value="capacity" v-on:change="updateValue($event.target.value)" @keydown.enter.prevent=""></input>\
-    </div></div>',
-    props: ['capacity'],
+    template: `<div>
```

Comment:
is this an extra div? doesn't seem used, not sure if it helps rendering somewhere else? (same message as below)

Reply:
No, it groups up `form-error `